sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -485,7 +485,65 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    * @since 1.4.0
    */
   def saveAsTable(tableName: String): Unit = {
-    saveAsTable(df.sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName))
+    import df.sparkSession.sessionState.analyzer.{AsTableIdentifier, CatalogObjectIdentifier}
+    import org.apache.spark.sql.catalog.v2.CatalogV2Implicits._
+
+    val session = df.sparkSession
+
+    session.sessionState.sqlParser.parseMultipartIdentifier(tableName) match {
+      case CatalogObjectIdentifier(Some(catalog), ident) =>
+        saveAsTable(catalog.asTableCatalog, ident, modeForDSV2)
+
+      case AsTableIdentifier(tableIdentifier) =>
```

Comment:
This should check whether the provider is v2 and use v2 and the V2SessionCatalog if it is.

```scala
case CatalogObjectIdentifier(None, ident) if isV1(provider) =>
  saveAsTable(v2SessionCatalog, ident, modeForDSV2)
```

Reply:
we separated the session catalog to a separate PR to get this in and add more tests to the session catalog support

Reply:
Sounds good to me.

Reply:
Sounds fine to me, but let's file a JIRA and mark it as a blocker for 3.0.

Reply:
Created SPARK-28666

Reply:
Could you add a TODO comment to this as well?